src/sources/aws_s3/mod.rs

```diff
@@ -290,6 +298,24 @@ mod integration_tests {
         test_event(key, Some("gzip"), None, None, buffer, logs).await;
     }
 
+    #[tokio::test]
+    async fn s3_process_message_multipart_gzip() {
```

Comment:
Any thoughts on adding a multipart zstd test as well?

Reply:
👍 I added in 4dac43502